programs/fileio.c

```diff
@@ -1332,6 +1332,7 @@ FIO_compressZstdFrame(FIO_ctx_t* const fCtx,
     FILE* const dstFile = ress.dstFile;
     U64 compressedfilesize = 0;
     ZSTD_EndDirective directive = ZSTD_e_continue;
+    size_t pledgedSrcSize = ZSTD_CONTENTSIZE_UNKNOWN;
```

Comment:
`size_t` is a memory-related length, expressed as a 32-bit value on 32-bit systems, and 64-bit on 64-bit systems.

However, `ZSTD_CONTENTSIZE_UNKNOWN` is always a 64-bit value.

There is a type mismatch on 32-bit.

I believe you want a 64-bit type for `pledgedSrcSize`, as it's the type expected by the `zstd` API for this field.

Reply:
Oh, okay I was inferring `size_t` from `prefs->streamSrcSize` being a `size_t`, but given that `fileSize` is a `U64` I see that it makes sense to go with the wider type. Will fix.